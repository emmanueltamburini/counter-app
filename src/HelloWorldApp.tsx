const newMessage = {
    message: "Hello World",
    title: "Fernando"
};

const helloWorld = () => "Hello world";

const HelloWorldApp = () => {
  return (
    <>
        <h1>{helloWorld()}</h1>
        <code>{JSON.stringify(newMessage)}</code>
        <p>I am a subtitle</p>
    </>
  )
}

export default HelloWorldApp;