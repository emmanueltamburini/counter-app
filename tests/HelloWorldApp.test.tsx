import React from "react";
import { render } from "@testing-library/react"
import HelloWorldApp from '../src/HelloWorldApp';

describe('HelloWorldApp test', () => {
    test('should be equal to snapshot', () => {
        const {container} = render(<HelloWorldApp name="Emmanuel" title="Title" subTitle={2}/>);
        expect(container).toMatchSnapshot();
    });

    test('should show an H1 element', () => {
        const title = 'Title';
        const subTitle = 2;
        const {container, getByText, getByTestId, getAllByText} = render(<HelloWorldApp name="Emmanuel" title={title} subTitle={subTitle}/>);
        expect(getByText(title)).toBeTruthy();
        expect(getByTestId('test-title').innerHTML).toContain(title);
        expect(getByTestId('test-subTitle').innerHTML).toContain((subTitle + 1).toString());
        expect(getAllByText((subTitle + 1).toString()).length).toBe(2);


        const h1 = container.querySelector('h1');
        expect(h1?.innerHTML).toContain(title);
    });


})