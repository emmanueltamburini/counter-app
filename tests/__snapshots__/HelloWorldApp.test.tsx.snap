// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorldApp test should be equal to snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Title
     
  </h1>
  <h2
    data-testid="test-subTitle"
  >
    3
  </h2>
  <h2
    data-testid="test-subTitle2"
  >
    3
  </h2>
  <code>
    {"message":"Hello World","title":"Fernando"}
  </code>
  <p>
    Hello Emmanuel
  </p>
</div>
`;
